@here/harp-text-canvas/lib/TextCanvas.ts

```diff
@@ -635,8 +635,8 @@ export class TextCanvas {
                 characterBounds = [];
             }
             if (params.storeStyles === true) {
-                renderStyle = this.m_currentTextRenderStyle;
-                layoutStyle = this.m_currentTextLayoutStyle;
+                renderStyle = this.m_currentTextRenderStyle.clone();
```

Comment:
Here TextBufferObject was holding shallow copies (references) all the time, even if TextCanvas.m_currentTextRenderStyle, TextCanvas.m_currentTextLayoutStyle is constantly changing!